### PR TITLE
test: e2e test for timeout of ics20 packet

### PR DESCRIPTION
**Pull request by seantking**
## Description

closes: #1975 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/transfer_test.go
Comment (seantking) [inline e2e/transfer_test.go:39]: @colin-axner I tried to use your [helper function](https://github.com/cosmos/ibc-go/blob/acda77b44fd9be710bb498a5c033eaba27df8f12/e2e/transfer_test.go#L31) for broadcasting the message instead of using the built-in library fn but for some reason the packet never timesout. 

Not sure if this is a golang relayer issue or if the message doesn't get broadcast correctly.

Comment (chatton) [reply e2e/transfer_test.go:39]: should we create an issue to investigate this?

File: e2e/transfer_test.go
Comment (colin-axner) [inline e2e/transfer_test.go:77]: I think this can be removed after merging main?

File: e2e/transfer_test.go
Comment (colin-axner) [reply e2e/transfer_test.go:39]: Agreed, I'd like to understand why this is the case?

**Review by colin-axner — COMMENTED** [review]
LGTM, will drop an approval after merge conflicts are fixed

File: .github/workflows/e2e-manual.yaml
Comment (damiannolan) [inline .github/workflows/e2e-manual.yaml:15]: Should this be duplicated? I guess we can remove, must be from merging main.

File: e2e/transfer_test.go
Comment (damiannolan) [inline e2e/transfer_test.go:133]: nit: should we make "IBC transfer packet timeout" or "IBC transfer packet times out"

File: e2e/transfer_test.go
Comment (damiannolan) [inline e2e/transfer_test.go:137]: same difference I think

```suggestion
		time.Sleep(time.Nanosecond) // want it to timeout immediately
```

**Review by damiannolan — APPROVED** [review]
Nice work, left some minor nits. Otherwise lgtm! 🚀

**Review by colin-axner — APPROVED** [review]
ACK nice work. I'm fine leaving the `SendIBCTransfer` usage, but it would be nice if the message was broadcasted, especially since we rely on the timeout creation by the cli

**Pull request merged**
